logviewer/api/transport_linux.go

```diff
@@ -0,0 +1,81 @@
+// build +linux
+
+package api
+
+import (
+	"context"
+	"fmt"
+	"golang.org/x/sys/unix"
+	"net"
+	"os"
+	"runtime"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const titusInits = "/var/lib/titus-inits"
+
+// nsDialer is a net.Dialer that does a Setns() into a container's network namespace before doing a connect
+type nsDialer struct {
+	containerNetNsPath string
+	systemDialer       net.Dialer
+}
+
+func (d *nsDialer) DialContext(ctx context.Context, network, address string) (net.Conn, error) {
+	log.Debugf("nsDialer: DialContext: %s", d.containerNetNsPath)
+	curThreadNsPath := fmt.Sprintf("/proc/%d/task/%d/ns/net", os.Getpid(), unix.Gettid())
```

Comment:
Why not `/proc/self/ns/net`?

Reply:
Paranoia - I wasn't sure if the ns for the entire process could change, if a bunch of its threads were all doing `Setns()`.  Not sure if that's possible or not.  Regardless, if I go the per-thread ns route, `/proc/thread-self` makes more sense.